### ISSUE TITLE
Mypy ignores `# type: ignore[override]` on multi-line function definition

### ISSUE THREAD

**Issue by skatsaounis** [python/mypy#16943]
**Bug Report**

Mypy cannot successfully understand the `# type: ignore[override]` on function return values if the function definition is split in many lines due to compliance with max line length.

**To Reproduce**

```python
async def get_response(
        self, path: str, scope: Scope
    ) -> Response | FileResponse:  # type: ignore[override]
```

**Expected Behavior**

Success like the one-liner:

```python
async def get_response(self, path: str, scope: Scope) -> Response | FileResponse:  # type: ignore[override]
```

```bash
check: OK (1.27=setup[0.01]+cmd[1.26] seconds)
```

**Actual Behavior**

Error:

```bash
check: FAIL code 1 (1.26=setup[0.01]+cmd[1.25] seconds)
error: Return type "Coroutine[Any, Any, Response | FileResponse]" of "get_response" incompatible with return type "Coroutine[Any, Any, Response]" in supertype "StaticFiles"  [override]
error: Unused "type: ignore" comment  [unused-ignore]
```

**Your Environment**

- Mypy version used: `1.8`
- Mypy command-line flags: `check: commands[0]> mypy my-code/`
- Mypy configuration options from `pyproject.toml`:

```
[tool.mypy]
install_types = true
junit_xml = "../junit-mypy.xml"
non_interactive = true
plugins = [
  "pydantic.mypy"
]
strict = true
```
- Python version used: `3.10.12`

**Comment by erictraut** [comment]
A `# type: ignore` comment suppresses errors only on the single line in which it is found. This is the behavior mandated in PEP 484 and the typing spec, and mypy is correctly conforming to the spec in this regard.

If you want to suppress an error reported on a different line, you'll need to move the `# type: ignore` comment to that line.

**Comment by skatsaounis** [comment]
Hi @erictraut

In my example I tried to move the comment on any of the three lines and I received the same error:

Alt 1
```python
async def get_response(  # type: ignore[override]
        self, path: str, scope: Scope
    ) -> Response | FileResponse:
```
Alt 2
```python
async def get_response(
        self, path: str, scope: Scope  # type: ignore[override]
    ) -> Response | FileResponse:
```
Alt 3
```python
async def get_response(
        self, path: str, scope: Scope
    ) -> Response | FileResponse:  # type: ignore[override]
```

Based on your comment it seems that I misinterpret which is the correct line to add the ignore instruction. Could you please suggest me the correct syntax so as to ignore override error on ` -> Response | FileResponse` part of my code?

Thank you in advance :slightly_smiling_face:

**Comment by erictraut** [comment]
Could you provide a minimal, self-contained code sample that exhibits the problem — preferably in the [mypy playground](https://mypy-play.net/)?

**Comment by skatsaounis** [comment]
Hi again,

While playing with the playground and trying to reproduce all the different ways that my use case is failing, I found out that one particular is working. I was under the wrong impression that I had tried it before opening this issue.

```py
async def get_response(  # type: ignore[override]
        self, path: str, scope: Scope
    ) -> Response | FileResponse:
```

This is the way to ignore override errors when function header is expanding in many lines. I am mentioning it here in case anyone else in the future is confused like me. While the error message is about the return type two lines below, the ignore should be placed at the first line of the function header.

Thank you for your time @erictraut and sorry for the inconvenience.
